File: fsharp/Fizzy.fs
namespace Fizzy

module Fizzy =
  let fizzbuzz n =
    let x, y  = (n % 3, n % 5)

    if (x, y) = (0, 0) then
      "FizzBuzz"
    else if x = 0 then
      "Fizz"
    else if y = 0 then
      "Buzz"
    else
      n.ToString("d")

  let main =
    Array.iter (fun s -> printfn "%s" s) (Array.Parallel.map fizzbuzz [1 .. 101])

File: perl/fizzy.pl
#!/usr/bin/env perl
use strict;
use warnings;

use Parallel::ForkManager;

sub fizzy {
  my $n = shift @_;

  my ($a, $b) = ($n % 3 == 0, $n % 5 == 0);

  if ($a && $b) {
    return "FizzBuzz";
  }
  elsif ($a) {
    return "Fizz";
  }
  elsif ($b) {
    return "Buzz";
  }
  else {
    return "$n";
  }
}

sub main {
  my $manager = new Parallel::ForkManager;

  my @fbs = 1 .. 100;

  foreach my $i (0 .. 99) {
    $manager->start and next;
    @fbs[$i] = fizzy($i + 1);
    $manager->finish;
  };

  foreach my $fb (@fbs) {
    print "$fb\n";
  }
}

main unless caller;
